src/microbe_stage/SpawnSystem.cs

```diff
@@ -247,13 +261,19 @@ private void SpawnEntities(Vector3 playerPosition, Vector3 playerRotation, int e
             spawn cycle, the SpawnSystem attempts to spawn each given
             entity multiple times depending on the spawnFrequency.
             numAttempts stores how many times the SpawnSystem attempts
-            to spawn the given entity.
+            to spawn the given entity. Furthermore, each attempt gives a spawn chance
+            for each spawner (up to their type limit) to still favor diversity,
+            instead of one spawner taking up all the free spawning slots.
```

Comment:
Was the issue that the spawners were making so much of one thing that they were reaching the maximum of living entities, which kept iron from spawning in the cave? From testing this change clearly works, but I don't understand why.

Reply:
That could certainly be a thing. For a while now, I've wanted per entity spawn caps (that are more reasonable, for example I think if the 1000 entity limit fills up with cells the game will lag, a lot).

Reply:
That was not even the total cap, but the per-frame cap (toxyns were taking all slots, that's why decreasing their density led to iron spawning)

Reply:
Per-frame? As in the spawner has been adding 500 toxyns per frame? I certainly haven't observed that in game...

Reply:
No, the limit was much smaller (2, I think), so at each frame, 2 entities would be spawned (which makes sense, both to not have long spawning frame and to be able to spawn new things to adapt - although it could be raised shortly after new patch spawn). 

Yet, given that toxins had such a high spawning rate, they always spawned up to the frame spawn limit (they roughly had ~100 chances out of 500 (max cap) so 20% chances, for 50 spawn trials or so). And thus, spawners later in the list were not tested in this frame, and the list would be reset at the next frame.

I guess that another fix could have been to save the remaining spawner list for next frame to cycle it all.

Reply:
> For a while now, I've wanted per entity spawn caps

Capping toxins to 100, and having a separate cap for iron and marine snow chunks, is what I meant in my comment. This doesn't have anything to do with per-frame caps, which also probably could be problematic if the toxins take up all the per-frame spawns... this is addressed by: https://github.com/Revolutionary-Games/Thrive/pull/2138 (which is an incomplete PR to change the spawn system) which uses a round robin approach to spawn from a randomized item list.

Reply:
Indeed a per-entity cap is a fully independant question, related to the total cap which is not the reason for the issues fixed here (but I agree this could prove useful overall).

This PR was really about identifying their cause and fixing them within the current system as they really harm game experience in patches.

Reply:
I'm also going to take a look at the incomplete spawn change PR though, see if I feel like taking it up.